yt_dlp/extractor/atvat.py

```diff
@@ -1,74 +1,194 @@
+        #             "subclip_end": 5866.72,
+        #             "id": 3
+        #         }
+        #         ]
+        #     },
+        #     "secure_delivery": true,
+        #     "iat": 1631820772,
+        #     "nbf": 1631820472,
+        #     "exp": 1631821072
+        # }
+        #  https://vas-v4.p7s1video.net/4.0/getsources?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCIsImtpZCI6InhfYXR2In0.eyJjb250ZW50X2lkcyI6eyJ2LWNlOWNnbjFlNzBuNSI6W3sic3ViY2xpcF9zdGFydCI6MCwic3ViY2xpcF9lbmQiOjE0NjYuNjgsImlkIjowfSx7InN1YmNsaXBfc3RhcnQiOjE0NjYuNjgsInN1YmNsaXBfZW5kIjoyOTMzLjM2LCJpZCI6MX0seyJzdWJjbGlwX3N0YXJ0IjoyOTMzLjM2LCJzdWJjbGlwX2VuZCI6NDQwMC4wNCwiaWQiOjJ9LHsic3ViY2xpcF9zdGFydCI6NDQwMC4wNCwic3ViY2xpcF9lbmQiOjU4NjYuNzIsImlkIjozfV19LCJzZWN1cmVfZGVsaXZlcnkiOnRydWUsImlhdCI6MTYzMTgyMDc3MiwibmJmIjoxNjMxODIwNDcyLCJleHAiOjE2MzE4MjEwNzJ9.FPlLuHa_au2Pv9kG5KPaRy7Uf2uX6t4lnoNLJEZ9Rf4
     def _real_extract(self, url):
-        display_id = self._match_id(url)
-        webpage = self._download_webpage(url, display_id)
-        video_data = self._parse_json(unescapeHTML(self._search_regex(
-            r'var\splaylist\s*=\s*(?P<json>\[.*\]);',
-            webpage, 'player data', group='json')),
-            display_id)
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        json_string = self._search_regex(
+            r'<script id="state" type="text/plain">(.*)</script>',
+            webpage, "json_data")
+        json_data = json.loads(json_string)
 
-        first_video = video_data[0]
-        video_id = first_video['id']
-        video_title = dict_get(first_video, ('tvShowTitle', 'title'))
+        # "views": {
+        #     "default": {
+        #         "page": {
+        #             "contentResource": [
+        contentResource = json_data["views"]["default"]["page"]["contentResource"]
+        content_id = contentResource[0]["id"]
+        content_ids = [{"id": id, "subclip_start": content["start"], "subclip_end": content["end"]} for id, content in enumerate(contentResource)]
 
-        return {
-            '_type': 'multi_video',
-            'id': video_id,
-            'title': video_title,
-            'entries': (self._process_entry(entry) for entry in video_data),
+        iat = datetime.datetime.now()              # time of request
+        nbf = iat - datetime.timedelta(minutes=5)  # not before
+        exp = iat + datetime.timedelta(minutes=5)  # expire
+        payload = {
+            "content_ids": {
+                content_id: content_ids,
+            },
+            "secure_delivery": True,
+            "iat": int(iat.timestamp()),
+            "nbf": int(nbf.timestamp()),
+            "exp": int(exp.timestamp()),
         }
+        # equivalent command using the external module `jwt`, but we implemented specifically this function ourselfs
+        # jwt_token = jwt.encode(payload, _ENCRYPTION_KEY, headers={"kid": "x_atv"}, algorithm="HS256")
```

Comment:
Most comments (including this) are unnecessary. Anyone can see that we are creating the jwt token from the below code `jwt_token = jwt_encode_hs256(...)`. There is no need to explain it again

Reply:
removed